azbus/receiver.go

```diff
@@ -199,10 +215,38 @@ func (r *Receiver) elapsed(ctx context.Context, count int, total int, maxDuratio
 	return err
 }
 
-// RenewMessageLock renews the given messages peek lock, so it doesn't lose the lock and get re-added to the message queue.
+// elapsed emits 2 log messages detailing how long processing took. Additionally executes suitable disposition of message.
+func (r *Receiver) elapsedParallel(ctx context.Context, count int, maxDuration time.Duration, msg *ReceivedMessage, handler PHandler) {
+	now := time.Now()
+
+	log := r.log.FromContext(ctx)
+	defer log.Close()
+
+	log.Debugf("Processing message %d", count)
+	disp, ctx, err := r.handleParallelReceivedMessageWithTracingContext(ctx, msg, handler)
+	_ = r.Dispose(ctx, disp, err, msg)
```

Comment:
we need to handle the error here.

Or we could set a suitable default for dispose if type of disposal is unknown, something like abandon or deadletter?

Happy with either

Reply:
the error is always nil

this is a hangover from the original design - once all service use the new pattern the returning of a nil error will be removed.

Reply:
The error isn't always nil, the default case has an `fmt.Errorf("unknown")` return?

I think we either need to change the default and then don't return error at all,

Or we need to handle the error, I would be surpised if our CI allows use to blank out errors like this?